src/VisualStudio/Core/Test.Next/Options/VisualStudioOptionStorageTests.cs

```diff
@@ -16,60 +16,158 @@ namespace Microsoft.CodeAnalysis.UnitTests;
 
 public class VisualStudioOptionStorageTests
 {
-    [Fact]
-    public void OptionValidation()
+    public static IEnumerable<object[]> ConfigNames
+    {
+        get
+        {
+            return OptionsTestInfo.CollectOptions(Path.GetDirectoryName(typeof(VisualStudioOptionStorage).Assembly.Location))
+                .Select(pair => new object[] { pair.Key });
+        }
+    }
+
+    public static IEnumerable<object[]> PerLanguageConfigNames
+    {
+        get
+        {
+            return OptionsTestInfo.CollectOptions(Path.GetDirectoryName(typeof(VisualStudioOptionStorage).Assembly.Location))
+                .Where(pair => pair.Value.Option.IsPerLanguage)
+                .Select(pair => new object[] { pair.Key });
+        }
+    }
+
+    public static IEnumerable<object[]> PublicOptionConfigNames
+    {
+        get
+        {
+            return OptionsTestInfo.CollectOptions(Path.GetDirectoryName(typeof(VisualStudioOptionStorage).Assembly.Location))
+                .Where(pair => pair.Value.Accessors.Any(a => a.option.PublicOption is not null))
+                .Select(pair => new object[] { pair.Key });
+        }
+    }
+
+    public static IEnumerable<object[]> ConfigNamesWithRoamingProfileStorage
+    {
+        get
+        {
+            return OptionsTestInfo.CollectOptions(Path.GetDirectoryName(typeof(VisualStudioOptionStorage).Assembly.Location))
+                .Where(pair => VisualStudioOptionStorage.Storages.TryGetValue(pair.Key, out var storage) && storage is VisualStudioOptionStorage.RoamingProfileStorage)
+                .Select(pair => new object[] { pair.Key });
+        }
+    }
+
+    public static IEnumerable<object[]> StorageNames
+    {
+        get
+        {
+            return VisualStudioOptionStorage.Storages
+                .Select(pair => new object[] { pair.Key });
+        }
+    }
+
+    /// <summary>
+    /// Options with per-language values shouldn't be defined in language-specific assembly since then they wouldn't be
+    /// applicable to the other language.
+    /// </summary>
+    [Theory]
+    [MemberData(nameof(PerLanguageConfigNames))]
```

Comment:
>PerLanguageConfigNames [](http://example.com/codeflow?start=23&length=22)

Not sure I see the benefit of parameterizing like this.

Reply:
1. Makes the failure easier to understand.
2. Allows directly stepping into the failure using the Debug feature of test explorer without the noise of any other option.

Reply:
I don't think [4] works for dynamic options.

Reply:
> I don't think [4] works for dynamic options.

That is unfortunately correct. I may update this again in the future for improved clarity.